feat: add foundational a11y documentation

josedialpad opened the pull request:
Jira ticket: https://switchcomm.atlassian.net/browse/DT-162

## Description
Added foundational a11y documentation to site.
New pages:
- Overview.
- Best Practices.
- Tooling.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [ ] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Brad Paugh, David Becher, or Drew Chandler.

## Obligatory GIF (super important!)
![gif](https://i.imgur.com/gP9sN.gif)

dialpad-drew commented on docs/_data/site-nav.json:
```diff
@@ -550,6 +550,27 @@
           "publish": "yes"
         }
       ]
+    },
+    {
+      "title": "Accessibility",
+      "url": "/accessibility/overview",
+      "pages": [
+        {
+          "title": "Overview",
+          "url": "/accessibility/overview/",
+          "publish": "yes"
+        },
+        {
+          "title": "Best Practices",
+          "url": "/accessibility/best-practices/",
+          "publish": "yes"
+        },
+        {
+          "title": "Tooling",
+          "url": "/accessibility/tooling/",
+          "publish": "yes"
+        }
+      ]
```
Can we avoid this as a top-level nav item and instead nest it under "Getting Started?"
<img width="1144" alt="Screen Shot 2021-11-29 at 7 30 27 PM" src="https://user-images.githubusercontent.com/77751297/143975086-34233b2c-4696-4f76-8aa3-6133b5a50c44.png">

josedialpad replied:
Moved under the "Getting Started" section https://github.com/dialpad/dialtone/pull/512/commits/ae1260e44367b18b58c952cd2496ed723a06174b.

In the screenshot, "Fundamentals" is the first page for accessibility but we have "Overview" in the google doc. Which one is preferable?

dialpad-drew replied:
Actually, maybe "Fundamentals." Good call!

josedialpad replied:
Ok, changed page to "Fundamentals"  and added "Overview" as a header of the first section https://github.com/dialpad/dialtone/pull/512/commits/c2932c081b2c7f997639f8f7ea0a35a16046beec

dialpad-drew commented on docs/accessibility/overview.html:
```diff
@@ -0,0 +1,94 @@
+  <li>Robust</li>
+  {% endol %}
+  {% header "h3", "Perceivable" %}
+  {% paragraph %}The interface and its content is designed and implemented in such a way that it works across a variety of users agents and assistive technologies.{% endparagraph %}
+  {% paragraph %}<strong>Examples:</strong>{% endparagraph %}
+  {% ul %}
+  <li>The handling of media content like video, audio and imagery</li>
+  <li>The handling of text alternatives through things like captioning and sign language</li>
+  <li>The handling of color and contrast</li>
+  <li>The control of text size and audio controls</li>
+  <li>The handling content interactions i.e. hover and focus states</li>
+  {% endul %} 
+  {% header "h3", "Operable" %}
+  {% paragraph %}Users are able to interact with and navigate the interface and its content.{% endparagraph %}
+  {% paragraph %}<strong>Examples:</strong>{% endparagraph %}
+  {% ul %}
+  <li>Keyboard operability</li>
+  <li>Time allowance</li>
+  <li>Seizures and other involuntary physical reactions</li>
+  <li>Navigation through things like focus order, page titles, links, headings and labels</li>
+  <li>Input methods like touch targets sizing, alternative input methods</li>
+  {% endul %}
+  {% header "h3", "Understandable" %}
+  {% paragraph %}The interface and its content use plain language that’s easy to comprehend. Even if users can perceive and navigate content, what if they can’t actually understand the information?{% endparagraph %}
+  {% paragraph %}<strong>Examples:</strong>{% endparagraph %}
+  {% ul %}
+  <li>Predictable and consistent navigation</li>
+  <li>Use of language such as abbreviations, jargon, reading levels</li>
+  <li>Predictable and consistent actionable components</li>
+  <li>Feedback and error handling</li>
+  <li>Help, such as contextual help and error prevention</li>
+  {% endul %}
+  {% header "h3", "Robust" %}
+  {% paragraph %}The interface and its content is designed and implemented in such a way that it works across a variety of users agents and assistive technologies.{% endparagraph %}
+  {% paragraph %}<strong>Examples:</strong>{% endparagraph %}
+  {% ul %}
+  <li>Parsing and generally ‘clean’ code like proper heading hierarchy, no duplicate attributes, complete start and end tags for elements</li>
+  <li>Name, role and value</li>
+  <li>Status messages, sending messaging to the user without interrupting their workflow by giving focus to an element. e.g. “added to cart” while shopping online should act as a message and not a redirection of activity and focus for the user</li>
+  {% endul %}
```
Can we "chunk up" the POUR sections a little better with better spacing? It's kinda hard to tell them apart.
<img width="686" alt="Screen Shot 2021-11-29 at 7 26 55 PM" src="https://user-images.githubusercontent.com/77751297/143975202-377ec1c5-dbf3-4b1f-9c17-f24b1d2a9d34.png">

josedialpad replied:
Updated the spacing of the principles https://github.com/dialpad/dialtone/pull/512/commits/c8dc9f0fd5fca39f3b69122459abec39584a2676

dialpad-drew commented on docs/accessibility/tooling.html:
```diff
@@ -0,0 +1,45 @@
+  {% header "h2", "Browser extension" %}
+  {% header "h3", "Chrome and Firefox" %}
+  <a href="https://wave.webaim.org/extension/" class="d-link" target="_blank">WebAim - WAVE Browser Extensions</a>
+</section>
+<section class="d-stack16">
+  {% header "h2", "Checklists" %}
+  {% header "h3", "Android Accessibility Checklist" %}
+  <a href="https://dequeuniversity.com/assets/pdf/module-android/module-android-checklist.pdf" class="d-link" target="_blank">Deque - Summary and Checklist: Android Native Application Accessibility</a>
+  {% header "h3", "iOS Accessibility Checklist" %}
+  <a href="https://dequeuniversity.com/assets/pdf/module-ios/module-ios-checklist.pdf" class="d-link" target="_blank">Deque - Summary and Checklist: iOS Application Accessibility</a>
+  {% header "h3", "Web Accessibility Checklist" %}
+  <a href="https://dequeuniversity.com/checklists/web/" class="d-link" target="_blank">Deque - Web Accessibility Checklist</a>
+</section>
+<section class="d-stack16">
+  {% header "h2", "Figma plugins" %}
+  {% header "h3", "Adee Comprehensive Accessibility Tool" %}
+  <a href="https://www.figma.com/community/plugin/931280467863251825/Adee-Comprehensive-Accessibility-Tool" class="d-link" target="_blank">Figma Plugins - Adee Comprehensive Accessibility Tool</a>
+  {% header "h3", "Stark" %}
+  <a href="https://www.figma.com/community/plugin/732603254453395948/Stark" class="d-link" target="_blank">Figma Plugins - Stark</a>
+</section>
+<section class="d-stack16">
+  {% header "h2", "Screen Readers" %}
+  {% paragraph %}To learn more about the various combinations of operating systems and screen reader software, read <a href="https://webaim.org/projects/screenreadersurvey9/" class="d-link" target="_blank">WebAim’s 2021 Screen Reader User Survey</a>.{% endparagraph %}
+  {% header "h3", "VoiceOver for Apple Computers" %}
+  <a href="https://www.apple.com/voiceover/info/guide/_1131.html" class="d-link" target="_blank">Apple - Appendix A: Commands and Gestures</a>
+  {% header "h3", "Narrator Shortcuts for Windows" %}
+  <a href="https://support.microsoft.com/en-us/windows/windows-keyboard-shortcuts-for-accessibility-021bcb62-45c8-e4ef-1e4f-41b8c1fc87fd" class="d-link" target="_blank">Microsoft - Appendix B: Narrator keyboard commands and touch gestures</a>
+  {% header "h3", "NVDA for Windows" %}
+  {% paragraph %}This is free software that can be installed on any Windows machine.{% endparagraph %}
+  {% paragraph %}<a href="https://www.microsoft.com/en-us/p/nvda-nonvisual-desktop-access/9nvl6z0tm57d?activetab=pivot:overviewtab" class="d-link" target="_blank">Microsoft - NVDA (NonVisual Desktop Access)</a>{% endparagraph %}
+  {% paragraph %}<a href="https://webaim.org/articles/nvda/" class="d-link" target="_blank">WebAim - Using NVDA to Evaluate Web Accessibility</a>{% endparagraph %}
+  {% header "h3", "JAWS for Windows" %}
+  {% paragraph %}This software is not free and costs from $95/year for home use.{% endparagraph %}
+  {% paragraph %}<a href="https://www.freedomscientific.com/products/software/jaws/" class="d-link" target="_blank">Freedom Scientific - JAWS</a>{% endparagraph %}
+  {% paragraph %}<a href="https://webaim.org/articles/jaws/" class="d-link" target="_blank">WebAim - Using JAWS to Evaluate Web Accessibility</a>{% endparagraph %}
+  {% header "h3", "VoiceOver on iOS" %}
+  {% paragraph %}<a href="https://support.apple.com/en-us/HT210076" class="d-link" target="_blank">Apple - Turn on and practice VoiceOver on iPhone</a>{% endparagraph %}
+  {% header "h3", "Talkback for Android" %}
+  {% paragraph %}<a href="https://support.google.com/accessibility/android/answer/6007100?hl%3Den" class="d-link" target="_blank">Google - Turn TalkBack on or off</a>{% endparagraph %}
+</section>
```
I understand this is he we handle `h2` and `h3` tags, visually this page is a lot, both the table of contents as well as all the deep link icons. Anything we can do about that?
<img width="1200" alt="Screen Shot 2021-11-29 at 7 27 26 PM" src="https://user-images.githubusercontent.com/77751297/143975403-4f750be3-0f43-467c-ac84-62d09b0d4ab5.png">

Something more like this?
<img width="1145" alt="Screen Shot 2021-11-29 at 7 28 38 PM" src="https://user-images.githubusercontent.com/77751297/143975424-38bed597-c8d5-40ff-8077-0f9614d56629.png">

josedialpad replied:
Yeah, using raw `h3` without defined `id` we can skip the link and the TOC plugin https://github.com/dialpad/dialtone/pull/512/commits/1a1b40ab6df869a9934f62f0ee5485b9b19e7042

dialpad-drew commented on docs/accessibility/overview.html:
```diff
@@ -0,0 +1,94 @@
+  {% paragraph %}Dialpad strives to maintain <strong>WCAG 2.1 AA</strong> compliance in our digital products.{% endparagraph %}
+  {% paragraph %}Everyone benefits when things are designed accessibly. Simply put: building products for accessibility is simply building products for better usability. Considering <strong>1 in 4</strong> US adults -- that’s 61 million people! -- live with a permanent disability and many more will have temporary or situational throughout their lifetime, accessibility isn’t for just some people.  ...
+  <div class="d-ta-center">
+    <img src="/assets/images/accessibility-graphic.png" alt="Accessibility graphic" />
```
Can we have slightly more useful alt text here?

josedialpad replied:
Updated using the alt text from the source of the image https://github.com/dialpad/dialtone/pull/512/commits/3e080652509866349b69db836e29a195242f2d4d

dialpad-drew commented:
Hey dude, this is good! I added some feedback and made a couple super small changes in the Google Doc, which should be highlighted.

josedialpad commented:
> Hey dude, this is good! I added some feedback and made a couple super small changes in the Google Doc, which should be highlighted.

thanks for reviewing, I've updated the PR

dialpad-drew reviewed (approved):
Approved with one small comment. Nice work, dude! 🙌
